lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -489,10 +488,16 @@ private StateIndication checkSuspend(StateIndication stateChange) {
 
 		/* FIXME: we might want to limit this behaviour to only a specific
 		 * set of error codes */
-		if(pendingConnect != null && checkConnectivity()) {
+		if(pendingConnect != null) {
+			if (!Hosts.isFallback(pendingConnect.host)) {
+				if (!checkConnectivity()) {
+					return new StateIndication(ConnectionState.failed, new ErrorInfo("connection failed", 80000));
```

Comment:
Not sure I understand why we move to a failed state, unless there is an unrecoverable error, the connection should always stay in the suspended state and keep retrying.